ceph_deploy/util/init_systems.py

```diff
@@ -0,0 +1,246 @@
+from ceph_deploy.lib import remoto
+
+# Provide mappings/translations for services depending
+# on Init system.  All services coming into the classes
+# here should be one of ceph-mon, ceph-osd, ceph-mds, or
+# ceph-radosgw. It's then up to the code here to Do The
+# Right Thing.
+SYSV_SERVICES = {
+    'ceph-mon': 'mon',
+    'ceph-mds': 'mds'
+}
+
+UPSTART_SERVICES = {
+    'ceph-radosgw': 'radosgw'
+}
+
```

Comment:
@osynge You are absolutely right.  There are a couple of things.  First, what I've done is slightly confusing because in the sysvinit case I've conflated a service with the ceph init scripts capability to take an optional daemon type/instance.

For example, there are really two ceph _services_, `ceph` and `ceph-radosgw`.  However, the `ceph` service (init script) can also take a daemon to act on (like `mds` `osd` `mon`).  So both of these are valid:

`service ceph start`
`service ceph start mon.hostname`

The goal with `SYSV_SERVICES` here is that when the function sees a request to "start ceph-mon", it knows to create a command along the lines of `service ceph start mon.<x>`.  This behavior is init system specific, not distro specific.  The `SYSV_SERVICES` dictionary name does not make this behavior clear at all.

On top of that, the Debian-derived _distros_ (*.deb* packages) use a `radosgw` service instead of `ceph-radosgw`, and we correctly capture that Ubuntu (Upstart) but not on Debian today (http://tracker.ceph.com/issues/12170).  The work I've done here would perpetuate that error.

As it turns out, today is my last day with Red Hat.  I hope to see this PR through, but I'm not sure if I can fully address the issue in http://tracker.ceph.com/issues/12170 and #318.  I need to rebase this to make it merge clean, but you are right that something else needs to happen.

I *think* it might be as simple as calling `self.remote_info.service_mapping()` in the places necessary in the needed places.  That would require an interface change.

What I'm getting at is that I"m not sure if I can get the change into this PR or not.  But I wanted to detail would be necessary to fix it.  I don't think this change takes any steps backwards, but it doesn't fix the known issue that you correctly point out.